pkg/cache/snapshot.go

```diff
@@ -84,12 +84,19 @@ func (c *Cache) Snapshot() Snapshot {
 		ResourceFlavors:          make(map[kueue.ResourceFlavorReference]*kueue.ResourceFlavor, len(c.resourceFlavors)),
 		InactiveClusterQueueSets: sets.New[string](),
 	}
+	cycleChecker := cycleChecker{cycles: make(map[string]bool, len(c.hm.Cohorts))}
```

Comment:
why do you need to check at this point?
I think the information should be stored in the cache somehow. We need to communicate it to the CQ and cohort statuses anyways.

Reply:
I'm storing this structure in the hierarchy.Manager (and thus cache) now, but it will still be called again during snapshotting - using the memoized result.